src/catalog/src/lib.rs

```diff
@@ -162,21 +176,218 @@ pub struct BootstrapArgs {
     pub bootstrap_role: Option<String>,
 }
 
-pub(crate) fn builtin_cluster_replica_config(bootstrap_args: &BootstrapArgs) -> ReplicaConfig {
-    ReplicaConfig {
-        location: ReplicaLocation::Managed {
-            size: bootstrap_args.builtin_cluster_replica_size.clone(),
-            availability_zone: None,
-            disk: false,
-        },
-        logging: default_logging_config(),
-        idle_arrangement_merge_effort: None,
+/// A read only API for the durable catalog state.
+#[async_trait]
+pub trait ReadOnlyDurableCatalogState {
+    /// Reports if the catalog state has been initialized.
+    async fn is_initialized(&self) -> Result<bool, Error>;
+
+    // TODO(jkosh44) add and implement open methods to be implementation agnostic.
+    /*   /// Checks to see if opening the catalog would be
+    /// successful, without making any durable changes.
+    ///
+    /// Will return an error in the following scenarios:
+    ///   - Catalog not initialized.
+    ///   - Catalog migrations fail.
+    async fn check_open(&self) -> Result<(), Error>;
+
+    /// Opens the catalog in read only mode. All mutating methods
+    /// will return an error.
+    ///
+    /// If the catalog is uninitialized or requires a migrations, then
+    /// it will fail to open in read only mode.
+    async fn open_read_only(&mut self) -> Result<(), Error>;*/
+
+    /// Returns the epoch of the current durable catalog state. The epoch acts as
+    /// a fencing token to prevent split brain issues across two
+    /// [`DurableCatalogState`]s. When a new [`DurableCatalogState`] opens the
+    /// catalog, it will increment the epoch by one (or initialize it to some
+    /// value if there's no existing epoch) and store the value in memory. It's
+    /// guaranteed that no two [`DurableCatalogState`]s will return the same value
+    /// for their epoch.
+    ///
+    /// None is returned if the catalog hasn't been opened yet.
+    ///
+    /// NB: We may remove this in later iterations of Pv2.
+    fn epoch(&mut self) -> Option<NonZeroI64>;
+
+    /// Returns the version of Materialize that last wrote to the catalog.
```

Comment:
What does None mean? What format is the string in? Is there any special "new" value (I don't remember if this used to exist or still does).

Reply:
> What does None mean?

`None` means that the catalog hasn't been initialized. I added a comment about this.

> What format is the string in?

Whatever format `semver::Version::to_string` returns. For some reason, this has always been serialized and deserialized at higher levels of the catalog. Maybe we should directly return a `semver::Version`?

> Is there any special "new" value (I don't remember if this used to exist or still does).
 Not that I'm aware of, this is either empty or set to some normal value once migrations are complete.